router/server.go

```diff
@@ -59,6 +59,10 @@ func NewRouter(ctx context.Context, h models.HandlerInterface, port int, g http.
 		Methods("POST", "DELETE")
 	gMux.Handle("/api/system/kubernetes/contexts", h.ProviderMiddleware(h.AuthMiddleware(http.HandlerFunc(h.GetContextsFromK8SConfig)))).
 		Methods("POST")
+	gMux.Handle("/api/system/kubernetes/contexts", h.ProviderMiddleware(h.AuthMiddleware(h.SessionInjectorMiddleware(h.ChangeK8sContext)))).
+		Methods("PUT")
```

Comment:
@utkarsh-pro how do we reconcile this against API design principle #1? https://docs.google.com/spreadsheets/d/1ABJCbfQRi0uN_YoP2kmHZ-lTI4S4QuvRk1_unomKNRE/edit#gid=1241442399

Reply:
We already had a `POST` method defined for the endpoint and it is also being used at some places. I had two options
1. Change endpoint: Didn't do so because that would mean that endpoint should be pointing to a new "resource", which it does not.
2. Add another method: Had to resort to this because the couldn't think of something better,

Any suggestions here would be helpful :sweat_smile:

Reply:
We might be looking at a /id

Reply:
`POST` to `/api/system/kubernetes/contexts/{name}`. Is this the suggestion?

Example:
`POST` to `/api/system/kubernetes/contexts/minikube`